frontend/src/components/file-explorer/TreeNode.test.tsx

```diff
@@ -1,148 +1,121 @@
 import React from "react";
-import { act, render } from "@testing-library/react";
+import { act } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { renderWithProviders } from "test-utils";
 import TreeNode from "./TreeNode";
-import { WorkspaceFile } from "#/services/fileService";
-
-const onFileClick = vi.fn();
-
-const NODE: WorkspaceFile = {
-  name: "folder",
-  children: [
-    { name: "file.ts" },
-    { name: "folder2", children: [{ name: "file2.ts" }] },
-  ],
-};
+import { selectFile, listFiles } from "#/services/fileService";
+
+vi.mock("../../services/fileService", async () => ({
+  listFiles: vi.fn(async (path: string = "/") => {
+    if (path === "/") {
+      return Promise.resolve(["folder1/", "file1.ts"]);
+    }
+    if (path === "/folder1/") {
+      return Promise.resolve(["file2.ts"]);
+    }
+    return Promise.resolve([]);
+  }),
+  selectFile: vi.fn(async () => Promise.resolve({ code: "Hello world!" })),
+  uploadFile: vi.fn(),
+}));
 
 describe("TreeNode", () => {
   afterEach(() => {
     vi.resetAllMocks();
   });
 
   it("should render a file if property has no children", () => {
-    const { getByText } = render(
-      <TreeNode
-        node={NODE}
-        path={NODE.name}
-        onFileClick={onFileClick}
-        defaultOpen
-      />,
+    const { getByText } = renderWithProviders(
+      <TreeNode path="/file.ts" defaultOpen />,
     );
 
     expect(getByText("file.ts")).toBeInTheDocument();
   });
 
-  it("should render a folder if property has children", () => {
-    const { getByText } = render(
-      <TreeNode
-        node={NODE}
-        path={NODE.name}
-        onFileClick={onFileClick}
-        defaultOpen
-      />,
+  it("should render a folder if it's in a subdir", async () => {
```

Comment:
TIP: If you only want to run a specific test, you can use the `.only` filter instead of the `.skip` on all other tests. E.g.:

```ts
it.only("should run this test and this test alone", () => {
  /* ... */
});
```